lib/clientutils.js

```diff
@@ -56,7 +56,14 @@ var updateState = function(state, newState, callback) {
 
 // process state (following plugin execution)
 var processState = function(r, callback) {
-  if (r.state.abortWithResponse) {
+  if (r.abort) {
```

Comment:
It would be nice to have some comments about the scenario each of these blocks is for so we remember down the line.

Reply:
Sure, I will add in a separate PR to prevent a nightmare rebase!